core/services/pipeline/common_http.go

```diff
@@ -38,6 +39,13 @@ func makeHTTPRequest(
 		return nil, 0, nil, 0, errors.Wrap(err, "failed to create http.Request")
 	}
 	request.Header.Set("Content-Type", "application/json")
+	if len(reqHeaders)%2 != 0 {
+		lggr.Warnw("Got odd number of headers, discarding last value", "reqHeaders", reqHeaders)
```

Comment:
It is easy to miss this warning, then debugging the missing header would be difficult.
I suggest we panic here, and inside http task we apply this condition and fail the task clearly.